File: lib/models/orderModel.dart
import 'cartModel.dart';

class orderModel {
  int? id;
  int? cartId;
  int? addressId;
  int? total;
  double? discount;
  int? discountAmount;
  int? shippingPrice;
  String? status;
  String? paymentMethod;
  String? paymentStatus;
  String? createdAt;
  String? updatedAt;
  Cart? cart;


  orderModel(
      {this.id,
        this.cartId,
        this.addressId,
        this.total,
        this.discount,
        this.shippingPrice,
        this.status,
        this.paymentMethod,
        this.discountAmount,
        this.paymentStatus,
        this.createdAt,
        this.updatedAt,
        this.cart});
  orderModel.fromJson(Map<String, dynamic> json) {
    id = json['id'] != null ? int.tryParse(json['id'].toString()) : null;
    cartId = json['cartId'] != null ? int.tryParse(json['cartId'].toString()) : null;
    addressId = json['addressId'] != null ? int.tryParse(json['addressId'].toString()) : null;
    total = json['total'] != null ? int.tryParse(json['total'].toString()) : null;
    discount = json['discount'] != null ? double.tryParse(json['discount'].toString()) : null;
    discountAmount = json['discountAmount'] != null ? int.tryParse(json['discountAmount'].toString()) : null;
    shippingPrice = json['shippingPrice'] != null ? int.tryParse(json['shippingPrice'].toString()) : null;
    status = json['status'];
    paymentStatus = json['paymentStatus'];
    paymentMethod = json['paymentMethod'];
    createdAt = json['createdAt'];
    updatedAt = json['updatedAt'];
    cart = json['Cart'] != null ? Cart.fromJson(json['Cart']) : null;
  }

  Map<String, dynamic> toJson() {
    final Map<String, dynamic> data = new Map<String, dynamic>();
    data['id'] = this.id;
    data['cartId'] = this.cartId;
    data['addressId'] = this.addressId;
    data['total'] = this.total;
    data['discount'] = this.discount;
    data['shippingPrice'] = this.shippingPrice;
    data['status'] = this.status;
    data['discountAmount'] = this.discountAmount;
    data['createdAt'] = this.createdAt;
    data['updatedAt'] = this.updatedAt;
    if (this.cart != null) {
      data['Cart'] = this.cart!.toJson();
    }
    return data;
  }
}

File: backend_code/models/orderModel.js
import sequelize from "../config/db_config.js";
import { DataTypes } from "sequelize";

const order = sequelize.define('Orders', { 
    id: { 
        type: DataTypes.INTEGER, 
        autoIncrement: true, 
        primaryKey: true, 
    },
    cartId: {   //in which user details can be retrived using foreign key of user used in cart model
        type: DataTypes.INTEGER, 
        allowNull: false, 
    }, 
    addressId: {   //address of user where order is to be delivered  ,can be differ from address set in profile
        type: DataTypes.INTEGER, 
        allowNull: false, 
    },
    total: { 
        type: DataTypes.INTEGER, 
        allowNull: false, 
    }, 
    discount: { 
        type: DataTypes.DOUBLE, 
        allowNull: true, 
    },
    shippingPrice: { 
        type: DataTypes.DOUBLE, 
        allowNull: false, 
    },
    status: {
         type: DataTypes.STRING, //on pending,send,approved,completed
         allowNull: false,
    },
    paymentMethod: {
        type: DataTypes.ENUM('cash', 'jazzcash', 'easypaisa', 'bankAccount'),
        allowNull: true, // Can be null initially
    },
    paymentStatus: {
        type: DataTypes.ENUM('cashOnDelivery', 'paid'),
        allowNull: false,
        defaultValue: 'cashOnDelivery'
    }
 },
 {
    timestamps:true
}
);

export default order;